### PR TITLE
Chore: Migrate ee/omnichannel folder to Typescript

### PR DESCRIPTION
**Pull request by reinaldonetof**
## How to test or reproduce

- Login with an account that has a role as `livechat-agent` or any role that can talk with clients from omnichannel
- Be `Available` or `Not Available` by the button
- Open the Queue List. Take a room.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/ee/omnichannel/containers/OmnichannelStatus.tsx [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
+interface IOmnichannelStatus {
+	searching: boolean;
+	goQueue: () => void;
+	queueSize: number;
+	inquiryEnabled: boolean;
+	user: IUser;
+}
+
+const OmnichannelStatus = memo(({ searching, goQueue, queueSize, inquiryEnabled, user }: IOmnichannelStatus) => {
+	if (!!searching || !(RocketChat.isOmnichannelModuleAvailable() && user?.roles?.includes('livechat-agent'))) {
```
Comment (diegolmello) [inline app/ee/omnichannel/containers/OmnichannelStatus.tsx:22]: It's required boolean, so there's no need for the `!!`.
```suggestion
	if (searching || !(RocketChat.isOmnichannelModuleAvailable() && user?.roles?.includes('livechat-agent'))) {
```

File: app/ee/omnichannel/lib/subscriptions/inquiry.ts [hunk 2]
```diff
@@ -53,23 +72,23 @@ export default function subscribeInquiry() {
...
 
-	connectedListener = RocketChat.onStreamData('connected', handleConnection);
-	queueListener = RocketChat.onStreamData(streamTopic, handleQueueMessageReceived);
+	connectedListener = sdk.onStreamData('connected', handleConnection);
+	queueListener = sdk.onStreamData(streamTopic, handleQueueMessageReceived);
 
 	try {
 		const { user } = store.getState().login;
-		RocketChat.getAgentDepartments(user.id).then(result => {
+		RocketChat.getAgentDepartments(user.id).then((result: { success: boolean; departments: ILivechatDepartment[] }) => {
```
Comment (diegolmello) [inline app/ee/omnichannel/lib/subscriptions/inquiry.ts:80]: You can use `sdk` here

File: app/ee/omnichannel/views/QueueListView.tsx [hunk 3]
```diff
@@ -19,18 +20,51 @@ import * as HeaderButton from '../../../containers/HeaderButton';
...
+interface INavigationOptions {
+	isMasterDetail: boolean;
+	navigation: CompositeNavigationProp<
+		StackNavigationProp<ChatsStackParamList, 'QueueListView'>,
+		StackNavigationProp<MasterDetailInsideStackParamList>
+	>;
+}
+
+interface IQueueListView extends INavigationOptions {
+	// TODO: Refactor when migrate getUserSelector
```
Comment (diegolmello) [inline app/ee/omnichannel/views/QueueListView.tsx:38]: Already done

**Review by diegolmello — CHANGES_REQUESTED** [review]
I'm unsure about using IRoom here.
Why not ISubscription?

**Comment by reinaldonetof** [comment]
> I'm unsure about using IRoom here. Why not ISubscription?

They are quite similar, but because of [this line](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3749/files#diff-93b8de2342ea43b6e336eb401d407de892f7f5e96d1f38308bd12d8d404e9e5dR98) I assumed that we are treating as IRoom instead of ISubscription. WDYT?

File: app/definitions/IRoom.ts [hunk 4]
```diff
@@ -55,13 +52,11 @@ export enum OmnichannelSourceType {
 	API = 'api',
 	OTHER = 'other' // catch-all source type
 }
-export interface IOmnichannelRoom extends Omit<IRoom, 'default' | 'featured' | 'broadcast' | ''> {
+export interface IOmnichannelRoom extends Partial<Omit<IRoom, 'default' | 'featured' | 'broadcast' | ''>> {
```
Comment (AlexAlexandre) [inline app/definitions/IRoom.ts:55]: I dont got why the `| ' ' > ` on the end of the line

**Review by AlexAlexandre — CHANGES_REQUESTED** [review]

File: app/definitions/IRoom.ts [hunk 5]
```diff
@@ -91,18 +88,22 @@ export interface IOmnichannelRoom extends Omit<IRoom, 'default' | 'featured' | '
...
 	lastMessage?: IMessage & { token?: string };
 
-	tags: any;
-	closedAt: any;
-	metrics: any;
-	waitingResponse: any;
-	responseBy: any;
-	priorityId: any;
-	livechatData: any;
+	tags?: any;
```
[the remark this answers is not in the document]
Comment (reinaldonetof) [reply app/definitions/IRoom.ts:91]: Sure, I changed only the tags: string[] and closedAt: Date, but the others, I don't know yet.

File: app/definitions/IRoom.ts (hunk 4 above)
Comment (reinaldonetof) [reply app/definitions/IRoom.ts:55]: Missclick. Fixed!

**Review by AlexAlexandre — APPROVED** [review]

**Pull request merged**